Event search result implementation with additional filters.

umavenki opened the pull request:
Event search result implementation with additional filters.

jgoldenberg29 reviewed (changes requested):
Looking good Uma. Well implemented on the different filters at the top of the page.

 I am waiting on Sisi to add my IP so I can get events and see how they are laid out. 
 
 At this point my only comment is the nav bar seems to be overlapping the filter buttons for most screen widths - anything over 425px wide seems to have an overlap. Let's create some buffer between the two elements at all sizes.

nihalerdal commented on src/Components/EventResult.jsx:
```diff
@@ -0,0 +1,188 @@
+                    {event.dates.startDate} • {event.dates.startTime}
+                  </Typography>
+                  <Typography variant="h6">{event.name}</Typography>
+                </CardContent>
```
If the event name is longer than one line, the grid layout gets misaligned. Maybe we can look into reducing the text size or adjusting the grid dimensions
 ![Screen Shot 2025-01-09 at 3 23 54 PM](https://github.com/user-attachments/assets/9206a23a-1168-4a18-a476-47f340f607ad)

umavenki replied:
Updated grid layout

nihalerdal commented on src/Components/EventResult.jsx:
```diff
@@ -0,0 +1,188 @@
+        ) : (
+          <Grid item xs={12}>
+            <Typography variant="body1">No events to display</Typography>
+          </Grid>
```
Seems like the EmptyList component isn't being called when the list is empty. Do you mind taking a look?

umavenki replied:
This is taken care, now its fixed to show empty event message.

umavenki replied:
Updated empty event message.

nihalerdal reviewed (commented):
Thanks Uma! Grid view looks great! Just a litte issue: 

- When the user searches with a filter selected, all results are displayed, but the filter button stays highlighted instead of switching to the 'All' button. 

I’m not sure if this is a blocker for approval, so I added it as a comment. We can check with Josh and either merge it or fix the issue and move forward

umavenki commented:
> Thanks Uma! Grid view looks great! Just a litte issue:
> 
> * When the user searches with a filter selected, all results are displayed, but the filter button stays highlighted instead of switching to the 'All' button.
> 
> I’m not sure if this is a blocker for approval, so I added it as a comment. We can check with Josh and either merge it or fix the issue and move forward

Thanks Nihal, I have fixed the issue and it should highlight 'All' button.

nihalerdal reviewed (approved):
So fast and looks perfect!
